Put visibility info closer to decision info (#3262)
github/docs: data/reusables/pages/new-or-existing-repo.md

> If your site is an independent project, you can create a new repository to store your site's source code. {% if currentVersion == "free-pro-team@latest" %}If you are using a {% data variables.product.prodname_free_user %} or a {% data variables.product.prodname_free_team %} for organizations account, this should be a public repository.{% endif %}
> 
> If your site is associated with an existing project, you can add the source code {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" or currentVersion == "github-ae@latest" %}to that project's repository, in a `/docs` folder on the default branch or on a different branch.{% else %}for your site to a `gh-pages` branch or a `docs` folder on the `master` branch in that project's repository.{% endif %} For example, if you're creating a site to publish documentation for a project that's already on {% data variables.product.product_name %}, you may want to store the source code for the site in the same repository as the project.

Reviewer: I think it's misleading to include this information here, because it implies that this restriction only applies to sites that are independent projects. In reality, folks attempting to publish a site in an existing repo will be unable to do so if the repo is private and they use Free.

```suggestion
If your site is an independent project, you can create a new repository to store your site's source code. If your site is associated with an existing project, you can add the source code {% if currentVersion == "free-pro-team@latest" or currentVersion ver_gt "enterprise-server@2.22" or currentVersion == "github-ae@latest" %}to that project's repository, in a `/docs` folder on the default branch or on a different branch.{% else %}for your site to a `gh-pages` branch or a `docs` folder on the `master` branch in that project's repository.{% endif %} For example, if you're creating a site to publish documentation for a project that's already on {% data variables.product.product_name %}, you may want to store the source code for the site in the same repository as the project.

{% if currentVersion == "free-pro-team@latest" %}If the account that owns the repository uses {% data variables.product.prodname_free_user %} or {% data variables.product.prodname_free_team %}, the repository must be public.{% endif %}
```
